clientgui/res/amdicon2.xpm

```diff
@@ -0,0 +1,308 @@
+/* XPM */
+static char *amdicon2_xpm[] = {
```

Comment:
These must be `static const char*` to avoid compiler warnings, even if it's against XPM spec.

Reply:
The compiler objected to what GIMP had written. I tweaked it to match what the first generation icons already had.

Reply:
Could you show the error message from compiler? The XPM data is made of string literals which are constant. Not having the data declared `const` generates [tons of warnings](https://travis-ci.org/BOINC/boinc/jobs/414841363#L2786). I think previously there was only one file not declaring the data `const`.

Reply:
GIMP exports the line as
`static char * D:\Sources_build\boinc_7_12_1\clientgui\res\androidicon2_xpm[] = {`

That's clearly not usable, and VS2013 responds with

Error	3	error C2143: syntax error : missing ';' before ':'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	4	error C2059: syntax error : ':'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	5	error C2017: illegal escape sequence	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	6	error C2146: syntax error : missing ';' before identifier 'boinc_7_12_1'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	7	error C4430: missing type specifier - int assumed. Note: C++ does not support default-int	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	8	error C2146: syntax error : missing ';' before identifier 'clientgui'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	9	error C2146: syntax error : missing ';' before identifier 'res'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr
Error	10	error C2146: syntax error : missing ';' before identifier 'androidicon2_xpm'	d:\sources_build\boinc_7-12-1\clientgui\res\androidicon2.xpm	2	1	boincmgr

I'll try again with const - I missed that specific point last time.

Yes, `static const char *androidicon2_xpm[] = {` is clean. I'll change it - I must have looked at the one file without `const` when fixing the first problem.

Reply:
Yes, it was the original Android icon (which was added later, and doesn't have the licence lines either). I'll fix that while I'm here. Actually, no - I'll pull in that flat colour Android2 icon, to deal with Vitali's objection at the same time.